runtime/src/eval/system.rs

```diff
@@ -372,54 +373,63 @@ pub fn call<'config, H: Handler>(
 
 	match handler.call(to.into(), transfer, input, gas, scheme == CallScheme::StaticCall, context) {
 		Capture::Exit((reason, return_data)) => {
-			runtime.return_data_buffer = return_data;
-			let target_len = min(out_len, U256::from(runtime.return_data_buffer.len()));
+			return save_return_value(runtime, reason, return_data, handler);
+		},
+		Capture::Trap(interrupt) => {
+			// push!(runtime, H256::default());
+			Control::CallInterrupt(interrupt)
+		},
+	}
+}
 
-			match reason {
-				ExitReason::Succeed(_) => {
-					match runtime.machine.memory_mut().copy_large(
-						out_offset,
-						U256::zero(),
-						target_len,
-						&runtime.return_data_buffer[..],
-					) {
-						Ok(()) => {
-							push_u256!(runtime, U256::one());
-							Control::Continue
-						},
-						Err(_) => {
-							push_u256!(runtime, U256::zero());
-							Control::Continue
-						},
-					}
-				},
-				ExitReason::Revert(_) => {
-					push_u256!(runtime, U256::zero());
+/// save return_value into parent runtime
+pub fn save_return_value<'config, H: Handler>(
+	runtime: &mut Runtime,
+	reason : ExitReason,
+	return_data : Vec<u8>,
+	_handler: & H
+	) -> Control<H> {
 
-					let _ = runtime.machine.memory_mut().copy_large(
-						out_offset,
-						U256::zero(),
-						target_len,
-						&runtime.return_data_buffer[..],
-					);
+	pop_u256!(runtime, out_offset, out_len);
+	try_or_fail!(runtime.machine.memory_mut().resize_offset(out_offset, out_len));
 
+	runtime.return_data_buffer = return_data;
+	let target_len = min(out_len, U256::from(runtime.return_data_buffer.len()));
+	match reason {
+		ExitReason::Succeed(_) => {
+			match runtime.machine.memory_mut().copy_large(
+				out_offset,
+				U256::zero(),
+				target_len,
+				&runtime.return_data_buffer[..],
+			) {
+				Ok(()) => {
```

Comment:
Why do you change the alignment of this code? If you save the alignment of the original code, the diff will be clear.